MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/AnimatedDashLineActivity.java

```diff
@@ -0,0 +1,188 @@
+        lineJoin(LINE_JOIN_ROUND)
+      );
+      mapboxMap.addLayer(animatedDashBikeLineLayer);
+    } catch (MalformedURLException malformedUrlException) {
+      Log.d("AnimatedDashLine", "Check the URL: " + malformedUrlException.getMessage());
+    }
+  }
+
+  private class RefreshDashAndGapRunnable implements Runnable {
+
+    private float valueOne, valueTwo, valueThree, valueFour, ValueFive;
+    private float dashLength = 1;
+    private float gapLength = 3;
+
+    // We divide the animation up into 40 totalNumberOfSteps to make careful use of the finite space in
+    // LineAtlas
+    private float totalNumberOfSteps = 40;
+
+    // A # of totalNumberOfSteps proportional to the dashLength are devoted to manipulating the dash
+    private float dashSteps = totalNumberOfSteps * dashLength / (gapLength + dashLength);
+
+    // A # of totalNumberOfSteps proportional to the gapLength are devoted to manipulating the gap
+    private float gapSteps = totalNumberOfSteps - dashSteps;
+
+    // The current currentStep #
+    private int currentStep = 0;
+
+    private String TAG = "AnimatedDashLine";
+
+    @Override
+    public void run() {
+      Log.d(TAG, "RefreshDashAndGapRunnable run: ");
+      currentStep = currentStep + 1;
+      if (currentStep >= totalNumberOfSteps) {
+        currentStep = 0;
+      }
+      if (currentStep < dashSteps) {
+        valueOne = currentStep / dashSteps;
+        valueTwo = (1 - valueOne) * dashLength;
+        valueThree = gapLength;
+        valueFour = valueOne * dashLength;
+        ValueFive = 0;
+      } else {
+        valueOne = (currentStep - dashSteps) / (gapSteps);
+        valueTwo = 0;
+        valueThree = (1 - valueOne) * gapLength;
+        valueFour = dashLength;
+        ValueFive = valueOne * gapLength;
+      }
+      Log.d(TAG, "RefreshDashAndGapRunnable run: here");
+
+      Float[] newFloatArray = new Float[] {valueTwo, valueThree, valueFour, ValueFive};
+
+      mapboxMap.getLayer("animated_line_layer_id").setProperties(
+        lineDasharray(newFloatArray));
+      Log.d(TAG, "RefreshDashAndGapRunnable run: layer done being gotten");
+      handler.postDelayed(this, animationSpeedMillseconds);
+    }
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+    handler.removeCallbacks(refreshDashAndGapRunnable);
+    refreshDashAndGapRunnable = null;
+    handler = null;
```

Comment:
I think you need to keep a reference to your runnable and cancel/set to null it in onDestroy()

Reply:
```suggestion
    handler = null;
```